backend/webserver/wittymail_server/emailapi.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+# coding=utf-8
+
+import smtplib
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText 
+from email.mime.base import MIMEBase 
+from email import encoders 
+import os
+import util.logger as logger
+
+log = logger.get_logger(__name__)
+
+# Boolean to make sure we do not try to send an email before login
+INITIALIZED = False
+
+def set_login_details(server, port, username, password):
+  assert server is not None and port is not None and username is not None and password is not None,\
+  log.error('server, port number, username or password is None')
+
+  # s is a global object used to interact with the SMTP server
+  global s
+
+  # Connect to the SMTP server and login
+  try:
+    s = smtplib.SMTP(server, port)
+    s.ehlo()
+    s.starttls() 
+    s.login(username, password) 
+
+  except Exception as e:
+     log.error('Incorrect server/user information provided')
+     return [-1, 'Incorrect server/user information provided']
```

Comment:
It could be a different error as well, like the server IP/hostname not reachable. Can you check if SMTPlib has different types of Exception (*Error) classes and write an except block for each?